trio/_tests/test_deprecate.py

```diff
@@ -11,23 +14,32 @@
 )
 from . import module_with_deprecations
 
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
 
 @pytest.fixture
-def recwarn_always(recwarn):
+def recwarn_always(recwarn: object) -> object:
     warnings.simplefilter("always")
     # ResourceWarnings about unclosed sockets can occur nondeterministically
     # (during GC) which throws off the tests in this file
     warnings.simplefilter("ignore", ResourceWarning)
     return recwarn
```

Comment:
idk if it matters, but could be `(recwarn: T) -> T:`

Reply:
Changed in [2a0a355](https://github.com/python-trio/trio/pull/2771/commits/2a0a355168b2ab542cdd85161b31bf21aaf7164b)